STM/src/main/java/org/jboss/stm/internal/optimistic/OptimisticLockRecord.java

```diff
@@ -135,7 +135,7 @@ public int topLevelCommit ()
     {       
         boolean stateOK = checkState();
         
-        if (stateOK)
+        if (!stateOK)
```

Comment:
LGTM - did you verify that the topLevelCommit does affect a rollback in this case?